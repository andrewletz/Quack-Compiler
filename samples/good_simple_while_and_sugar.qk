/**
 * Simple test method to verify that while loops work as expected.  It also
 * verifies the proper calling of the sugared compare commands.
 */
x : Int = 1;
while x < 10 {
    x.PRINT();
    """\n""".PRINT();
    x = x + 1;
}

x = 12;
while x <= 13 {
    x.PRINT();
    """\n""".PRINT();
    x = x + 1;
}

x : Int = 15;
while x > 10 {
    x.PRINT();
    """\n""".PRINT();
    x = x - 1;
}

x = 4;
while x >= 2 {
    x.PRINT();
    """\n""".PRINT();
    x = x - 1;
}

x = 0;
while not (x == 5) {
    x.PRINT();
    """\n""".PRINT();
    x = x + 1;
}

