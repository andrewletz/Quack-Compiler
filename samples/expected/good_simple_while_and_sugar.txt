1
2
3
4
5
6
7
8
9
12
13
15
14
13
12
11
4
3
2
0
1
2
3
4
